src/reporter.js

```diff
@@ -1,52 +1,74 @@
-const SMASHTEST_SS_DIR = path.join('smashtest', 'screenshots');
+const fs = require("fs");
+const path = require("path");
+const readFiles = require("read-files-promise");
+const mustache = require("mustache");
+const utils = require("./utils.js");
+const chalk = require("chalk");
+const getPort = require("get-port");
+const WebSocket = require("ws");
+
+let REPORT_FILENAME,
+    REPORT_DATA_FILENAME,
+    PASSED_DATA_FILENAME,
+    SMASHTEST_SS_DIR,
+    initialFolder,
+    folder;
+
+const date = Date.now();
 
 /**
  * Generates a report on the status of the tree and runner
  */
 class Reporter {
-    constructor(tree, runner) {
-        this.tree = tree;               // the Tree object to report on
-        this.runner = runner;           // the Runner object to report on
+  constructor(tree, runner) {
+    this.tree = tree; // the Tree object to report on
+    this.runner = runner; // the Runner object to report on
 
-        this.reportTemplate = "";       // template for html reports
-        this.reportTime = null;         // Date when the report was generated
+    this.reportTemplate = ""; // template for html reports
+    this.reportTime = null; // Date when the report was generated
 
-        this.isReportServer = true;     // whether or not to run the report server
-        this.reportDomain = null;       // domain:port where report server's api is available
-        this.wsServer = null;           // websocket server object
+    this.isReportServer = true; // whether or not to run the report server
+    this.reportDomain = null; // domain:port where report server's api is available
+    this.wsServer = null; // websocket server object
 
-        this.prevSnapshot = null;       // previous snapshot sent over websockets
+    this.prevSnapshot = null; // previous snapshot sent over websockets
 
-        this.timerFull = null;          // timer that goes off when it's time to do a full write
-        this.timerSnapshot = null;      // timer that goes off when it's time to do a snapshot write
+    this.timerFull = null; // timer that goes off when it's time to do a full write
+    this.timerSnapshot = null; // timer that goes off when it's time to do a snapshot write
 
-        this.stopped = false;           // true if this Reporter has been stopped
-    }
+    this.stopped = false; // true if this Reporter has been stopped
 
-    /**
-     * @return {String} The absolute path of the report html file
-     */
-    getFullReportPath() {
-        return path.join(process.cwd(), REPORT_FILENAME);
-    }
+    this.reportPath = "";
+    this.history = "false";
+  }
+
+  /**
+   * @return {String} The absolute path of the report html file
+   */
+  getFullReportPath() {
+    return path.join(folder, "report.html");
+  }
+
+  getPathFolder() {
+    initialFolder = (this.history === "false" ? "smashtest" : `report/smashtest-${date}`);
+    folder = (this.reportPath === "" ? initialFolder : `${this.reportPath}/${initialFolder}`);
```

Comment:
Ok now I understand the `--history` flag. Love it!

Reply:
I think this will render the folder as something like "smashtest-1570559597444". I kind of don't like the long number. Is it possible to format the folder like "smashtest-2019-10-08T14-34-11" (or something like that)?

Reply:
We should also be consistent in where smashtest-related output files go. They should always go in a `smashtest` folder in the directory from where the `smashtest` command was run. Inside, we have a file called `report.html`, or alternatively, have folders like `smashtest-[DATETIME]`, inside which you find `report.html`. I don't think we need a `report` folder.

Reply:
The report folder is other fonctionality (optional) if you need to change colaction for reporting.
You can --history=true and didn't set report folder.
For default folder is a smashtest folder in the directory from where the smashtest command was run.
But you can change it with `report-path`

Reply:
i change Date format for this exemple : `report\smashtest-2019-10-09T11-37-17\report.htm`